stdlib/statistics.pyi

```diff
@@ -76,7 +79,7 @@ def stdev(data: Iterable[_NumberT], xbar: _NumberT | None = None) -> _NumberT: .
 def variance(data: Iterable[_NumberT], xbar: _NumberT | None = None) -> _NumberT: ...
 
 class NormalDist:
-    def __init__(self, mu: float = 0.0, sigma: float = 1.0) -> None: ...
+    def __init__(self, mu: ConvertibleToFloat = 0.0, sigma: float = 1.0) -> None: ...
```

Comment:
should `sigma` not also be `ConvertibleToFloat` here, if `mu` is? It looks like they both have `float` called on them in `__init__`: https://github.com/python/cpython/blob/39b2f82717a69dde7212bc39b673b0f55c99e6a3/Lib/statistics.py#L913-L914

Reply:
Yea I wanted to do that actually, but then I noticed that it checks `if sigma < 0.0` before that:
https://github.com/python/cpython/blob/39b2f82717a69dde7212bc39b673b0f55c99e6a3/Lib/statistics.py#L911

Reply:
Ohh great point. Hmm, but it is a bit weird to have different types for `mu` and `sigma`, no? I guess I sorta lean towards leaving this annotation as-is if we can't change them both. It indicates to me that you're probably not _meant_ to pass in something that's not an `int` or a `float` to this function, even if it (accidentally?) works for the `mu` parameter. And `float` as an annotation already means "`int` or `float`"

Reply:
Yea I agree. I considered writing a `Protocol` with `__float__` and `__gt__: (float) -> ConvertibleToFloat` and use it on both sides, but thought that that might be a bit *too much* here. 
But now that I think about it, it might actually be better to do so after all. Because that way it would also accept numpy floats and integers. What do you think?

Reply:
is it common to use the `statistics` module with numpy floats and integers? My understanding was that the `statistics` module generally isn't what you'd want to use if performance is important to you (it prioritises simplicity of implementation), and that numpy/scipy/etc generally provided more optimised alternatives.

I might be wrong, though -- I've never done much scientific programming. You probably know more than me here!

Reply:
> is it common to use the `statistics` module with numpy floats and integers?

Well, I personally never have. But since NumPy itself doesn't provide a normal distribution class, I can imagine that there people might actually use it that way.

Reply:
I believe that `fractions.Fraction` would also work at runtime.

Reply:
I still worry that we might be overly encoding implementation details (that may only accidentally work currently) if we go with the protocol approach. If we had an actual use case, that would be a different question -- but it _sounds_ like this is purely theoretical, so I'm inclined to leave it as-is until somebody with a use case complains about it